src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random rand = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int number = rand.nextInt(5);
+        switch (number) {
+            case 1:
+                return new Circle(rand.nextInt(100),
+                        colorSupplier.getRandomColor());
```

Comment:
Avoid passing expressions directly into the constructor. Assign `rand.nextInt(100)` and `colorSupplier.getRandomColor()` to local variables before passing them to the `Circle` constructor for better readability.